# HG changeset patch
# User Edmund Yan <edmundhyan@gmail.com>
# Date 1346362537 25200
# Node ID 7f3d310e28b3eea2083fd025c377867d5180fda9
# Parent  3c06b724923a5ec6e4dd72e87e08c1b7d1d85294
Adding target name to --debug=time output

diff -r 3c06b724923a -r 7f3d310e28b3 src/engine/SCons/Script/Main.py
--- a/src/engine/SCons/Script/Main.py	Wed Aug 29 13:59:56 2012 -0400
+++ b/src/engine/SCons/Script/Main.py	Thu Aug 30 14:35:37 2012 -0700
@@ -187,7 +187,10 @@
             finish_time = time.time()
             last_command_end = finish_time
             cumulative_command_time = cumulative_command_time+finish_time-start_time
-            sys.stdout.write("Command execution time: %f seconds\n"%(finish_time-start_time))
+            if GetOption("verbose"):
+                sys.stdout.write("Command execution time: %s : %f seconds\n"%(str(self.node), finish_time-start_time))
+            else:
+                sys.stdout.write("Command execution time: %f seconds\n"%(finish_time-start_time))
 
     def do_failed(self, status=2):
         _BuildFailures.append(self.exception[1])
